src/main/resources/avro/common.avdl

```diff
@@ -17,31 +29,88 @@ error GAException {
+coordinates.
+*/
+enum Side {
+  PLUS,
+  MINUS
+}
+
+/**
+An abstraction for referring to a genomic position on a sequence.
+
+For now, represents a genomic position as a sequence ID, a base offset on that
+sequence (0-based), and a flag to say if it's the forward or reverse strand that
+we're talking about.
 */
-// TODO: Add support here for universally unique base IDs.
 record Position {
   /**
-  The name of the reference (or, more technically, the scaffold) in whatever
-  reference set is being used. Does not generally include a "chr" prefix, so for
-  example "X" would be used for the X chromosome.
+  The ID of the sequence on which the `Position` is located.
   */
-  string referenceName;
+  string id;
 
   /**
-  The 0-based offset from the start of the forward strand for that reference.
-  Genomic positions are non-negative integers less than reference length.
+  The 0-based offset from the start of the sequence string.
   */
-  long position;
+  long offset;
 
   /**
-  A flag to indicate if we are on the forward strand (`false`) or reverse
-  strand (`true`).
+  A flag to indicate if we are on the forward strand/downstream face (`PLUS`) or
+  reverse strand/upstream face (`MINUS`).
+  */
+  Side side;
+}
+
+/**
+A `Segment` represents a piece of sequence, which is optionally joined onto
+other `Segment`s of sequence at each end, forming a graph. The bases for a
+`Segment` are available through the `ListSegmentBases` API call.
+*/
+record Segment {
+  /**
+  The sequence ID and start index of this `Segment`. This base is always
+  included in the segment, regardless of strand.
+  */
+  Position start;
+
+  /**
+  The length of this `Segment`'s sequence, walking away from `start` in the
+  direction of `start.side`.
+  */
+  long length;
+
+  /**
+  Start and end `Position`s where this `Segment` attaches to other sequences.
+  Note that the segmentId for start and end might not be the same. The
+  `Segment`s covering the sequences onto which this `Segment` is joined are
+  called its "parents".
+  
+  `startJoin` is the join on the base at `start`, and may only be set if `start`
+  is at one end of the underlying sequence, pointing inwards. Similarly,
+  `endJoin` is the join at on the base at the opposite end of the `Segment`, and
+  may only be set if that base is similarly at one end of the underlying
+  sequence, being read in the appropriate direction.
+  */
+  union { null, Position } startJoin, endJoin;
+}
+
+/**
+A `Path` is an ordered list of `Segment`s. In general any contiguous path
+through a sequence graph, with no novel adjacencies, can be represented by a
+`Path`.
+*/
+record Path {
+  /**
+  We require that each adjacent pair of `Segment`s in a `Path` be related by a
+  join of one onto the end of the other that is present in the underlying
+  sequence graph. We also require that the whole array describes a single linear
+  path.
   */
-  boolean reverseStrand;
+  array<Segment> segments = [];
```

Comment:
On the one hand, it's a good point that we might want a really long `Path` for some `Allele`, and it might be a bit of a pain to download it all at once.

On the other hand, I anticipate the vast majority of `Allele`s being SNPs, with paths of only a few `Segment`s. Do we want to force an additional round trip to the server for every `Path`?

Either way, we can probably make that change after this gets accepted; it seems fairly straightforward.